### DISCUSSION TITLE
What is required to add a new printer to Orca Slicer?

### DISCUSSION THREAD

**Discussion by julie777** [OrcaSlicer/OrcaSlicer#1140] [Q&A]
I am in the process of learning enough about the code to start helping with some of the coding tasks. Given the flurry of activity caused by constantly adding support for additional printers I started wondering what is involved.

I hope that someone will be willing to help me get a better understanding.

I know that adding a printer is not as simple as just creating a printer profile and dropping it into the profiles directory. The information that I can think of that is needed is:
  - printer profile which is used to configure slicing
  - which firmware it is running to know what G code is valid
  - which flavor of the firmware that particular printer is running, because there are custom G codes
  - how to communicate with the printer to get status, send jobs, etc.

If I separate into areas of Orca Slicer
  - network communication
  - creating g code to print
  - custom gcode in profiles
  - user interaction with settings
I can see how different information is needed. For example, if a printer, like my Prusa, is not network capable then communication is not available.

Not everything is that obvious to me.

Is the base firmware/G code kind enough for the slicer to output a printable G code file? 
I know that all the basic movement and extrude G code is pretty much the same, but I don't know if any non-standard code is required.

Can all the printer specific G code be put in the various profile sections, or does the slicer have to know about it?

Is printer control separate from the actual networking code?
I know that different printer brands/models can use the same networking and job control some of the time. The octopi comes to mind. And Klipper on most if not all machines should be using the same API server.
In contrast, BBL uses multiple protocols for different types of network communication and tries to hide it all in closed source DLLs.

Is there a specific list of required information somewhere that someone can use as a checklist and provide all the information necessary to have their favorite printer model added to Orca Slicer?

**Comment by urself25** [comment]
Hi, I'm attempting to do the same. Were you able to find your answers?
